network/src/connectivity_manager/mod.rs

```diff
@@ -823,6 +839,68 @@ where
             }
         }
     }
+
+    async fn report_network_metrics(&self) {
```

Comment:
My biggest concern with this change is that we're starting to litter the codebase with telemetry operations. This method by itself is not bad, but if we'll eventually want to expose more telemetry data, this is going to get messy really quickly.

In a perfect world, we'd somehow leverage the metrics that we're expecting all components to emit already. For example, on each metric counter increment or gauge change, we'd have a hook to also execute the telemetry code to decide what information to forward via telemetry. It might be worth us exploring how to do this in the code (e.g., rust macros). Interested to see other peoples thoughts, here 😄

Reply:
Thanks @rustielin! Makes sense to have this ready to unblock us 😄 Just something for us to be aware of going forward (i.e., if we decide to add a lot more interesting data).